crates/router/src/core/blocklist/utils.rs

```diff
@@ -357,3 +365,255 @@ async fn delete_card_bin_blocklist_entry(
+                    merchant_fingerprint_secret.as_bytes(),
+                    card.card_number.clone().get_extended_card_bin().as_bytes(),
+                )
+                .attach_printable("error in extended card bin hash creation")
+                .map_or_else(
+                    |err| {
+                        logger::error!(error=?err);
+                        None
+                    },
+                    Some,
+                )
+            }
+            _ => None,
+        })
+        .map(hex::encode);
+
+    //validating the payment method.
+    let mut blocklist_futures = Vec::new();
+    if let Some(card_number_fingerprint) = card_number_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            merchant_id,
+            card_number_fingerprint,
+        ));
+    }
+
+    if let Some(card_bin_fingerprint) = card_bin_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            merchant_id,
+            card_bin_fingerprint,
+        ));
+    }
+
+    if let Some(extended_card_bin_fingerprint) = extended_card_bin_fingerprint.as_ref() {
+        blocklist_futures.push(db.find_blocklist_lookup_entry_by_merchant_id_fingerprint(
+            merchant_id,
+            extended_card_bin_fingerprint,
+        ));
+    }
+
+    let blocklist_lookups = futures::future::join_all(blocklist_futures).await;
+
+    let mut db_operations_successful = false;
+    for lookup in blocklist_lookups {
+        match lookup {
+            Ok(_) => {
+                db_operations_successful = true;
+            }
+            Err(e) => {
+                logger::error!(blocklist_db_error=?e, "failed db operations for blocklist");
+            }
+        }
+    }
+
+    if db_operations_successful {
+        // Update db for attempt and intent status.
+        db.update_payment_intent(
+            payment_data.payment_intent.clone(),
+            storage::PaymentIntentUpdate::RejectUpdate {
+                status: common_enums::IntentStatus::Failed,
+                merchant_decision: Some(MerchantDecision::Rejected.to_string()),
+                updated_by: merchant_account.storage_scheme.to_string(),
+            },
+            merchant_account.storage_scheme,
+        )
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::PaymentNotFound)
+        .attach_printable(
+            "Failed to update status in Payment Intent to failed due to it being blocklisted",
+        )?;
+
+        // If payment is blocked not showing connector details
+        let attempt_update = storage::PaymentAttemptUpdate::BlocklistUpdate {
+            status: common_enums::AttemptStatus::Failure,
+            error_code: Some(Some("HE-03".to_string())),
+            error_message: Some(Some(
+                "Failed to update status in Payment Attempt to failed, due to it being blocklisted"
```

Comment:
```suggestion
                "This payment was blocked"
```
This message will be displayed to the merchant, so have an appropriate error message.

Reply:
This payment method has been blocked